ledger-api/grpc-definitions/com/daml/ledger/api/v2/event_query_service.proto

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.daml.ledger.api.v2;
+
+import "com/daml/ledger/api/v1/event.proto";
+import "com/daml/ledger/api/v1/event_query_service.proto";
+
+option java_outer_classname = "EventQueryServiceOuterClass";
+option java_package = "com.daml.ledger.api.v2";
+option csharp_namespace = "Com.Daml.Ledger.Api.V2";
+
+// Query events by contract id or key.
+service EventQueryService {
+
+  // Get the create and the consuming exercise event for the contract with the provided ID.
```

Comment:
Not for now. It would require a completely different approach here with continuation tokens etc. I would like to suggest to postpone / spin off enhancing this endpoint, other than it is absolutely mandatory for TestNet.

Reply:
wfm